### PR TITLE
Inline NewCourse.vue into NewCourseModal.vue

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

This PR inlines NewCourse into NewCourseModal. This can help us avoid some awkward component communication via ref and also allows us to find some duplicated states that can be easily derived. e.g.

- left button text is always cancel when the course is not selected, always add when the course is selected.
- right button text is always next is edit mode, and add when not in edit mode.
- right button is only disabled when a course is not selected.
- goBack state is unused

### Test Plan 

Local:
local
Master:
master

### Notes

In a future PR, I plan to fix the correctness issue of the requirement choices by using the graph.

### CODE REVIEW

File: src/components/Modals/NewCourse/NewCourseModal.vue [hunk 1]
```diff
@@ -60,31 +95,74 @@ export default Vue.extend({
     isCourseModelSelectingSemester: Boolean,
     reqs: Array as PropType<readonly SingleMenuRequirement[]>,
```
Comment (tcho6319) [inline src/components/Modals/NewCourse/NewCourseModal.vue:96]: Can we do `type` and `required` for the props declarations?

File: src/components/Modals/NewCourse/NewCourseModal.vue [hunk 2]
```diff
@@ -4,51 +4,86 @@
     contentClass="content-course"
     :leftButtonText="leftButtonText"
     :rightButtonText="rightButtonText"
-    :rightButtonIsDisabled="isDisabled"
+    :rightButtonIsDisabled="selectCourse == null"
```
Comment (tcho6319) [inline src/components/Modals/NewCourse/NewCourseModal.vue:7]: Do you mean `selectedCourse == null`? I'm not sure if this is working properly. When I first open up the New Course Modal, the Add button should be grayed out. (Left is PR, right is master).
Screen Shot 2021-02-09 at 2 43 53 PM

**Review by tcho6319 — CHANGES_REQUESTED** [review]
Thanks Sam! I requested 2 smaller changes, but once those are addressed, the PR should be good for approval.

File: src/components/Modals/NewCourse/NewCourseModal.vue (hunk 1 above)
Comment (SamChou19815) [reply src/components/Modals/NewCourse/NewCourseModal.vue]: changed

File: src/components/Modals/NewCourse/NewCourseModal.vue (hunk 2 above)
Comment (SamChou19815) [reply src/components/Modals/NewCourse/NewCourseModal.vue]: should be fixed now

**Review by tcho6319 — APPROVED** [review]
LGTM! Thank you :)

**Pull request merged**